openedx/tests/xblock_integration/test_done.py

```diff
@@ -0,0 +1,233 @@
+"""
+This tests that the completion XBlock correctly stores state. This
+is a fairly simple XBlock, and a correspondingly simple test suite.
+"""
+
+import json
+import mock
+import unittest
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+
+from lms.djangoapps.courseware.tests.helpers import LoginEnrollmentTestCase
+from lms.djangoapps.courseware.tests.factories import GlobalStaffFactory
+
+
+class SimpleEventTestMixin(object):
```

Comment:
Also, you should be able to leverage this:

https://github.com/edx/edx-platform/blob/master/openedx/core/lib/tests/assertions/events.py

Which provides most of the flexible assertion magic that is available in the acceptance tests.